libraries/chain/controller.cpp

```diff
@@ -177,6 +177,19 @@ struct block_data_t {
    uint32_t             head_block_num()  const { return std::visit([](const auto& bd) { return bd.head->block_num(); }, v); }
    block_timestamp_type head_block_time() const { return std::visit([](const auto& bd) { return bd.head->timestamp(); }, v); }
    account_name         head_block_producer() const { return std::visit([](const auto& bd) { return bd.head->producer(); }, v); }
+
+   void transition_fork_db_to_if(const auto& vbsp) {
+      std::visit([](auto& bd) { bd.fork_db.close(); }, v);
+      auto bsp = std::make_shared<block_state>(*std::get<block_state_legacy_ptr>(vbsp));
+      v.emplace<block_data_new_t>(std::visit([](const auto& bd) { return bd.fork_db.get_data_dir(); }, v));
+      std::visit(overloaded{
+                    [](block_data_legacy_t&) {},
+                    [&](block_data_new_t& bd) {
+                       bd.head = bsp;
+                       bd.fork_db.reset(*bd.head);
```

Comment:
Do you need to re-open fork_db?

Reply:
No, nothing to read from disk and we are using the same path of the previous one. It would be a no-op. Although, might be clearer to call it anyway.